src/clients/mocks/MockEvents.ts

```diff
@@ -44,7 +58,12 @@ export class EventManager {
 
     let { blockNumber, transactionIndex } = inputs;
 
-    blockNumber = blockNumber ?? random(1, 100_000, false);
+    // Increment the block number by at least 1, by default. The caller may override
+    // to force the same block number to be used, but never a previous block number.
+    blockNumber = blockNumber ?? random(this.blockNumber + 1, this.blockNumber + this.minBlockRange, false);
+    assert(blockNumber >= this.blockNumber, `${blockNumber} < ${this.blockNumber}`);
```

Comment:
Should we leave a more explicit error? 
```suggestion
    assert(blockNumber >= this.blockNumber, `Inequality failed to match ${blockNumber} < ${this.blockNumber}`);
```

Reply:
I'd prefer to leave it as is - `Inequality failed to match ...` isn't adding any information that isn't already there, and when these errors throw there tends to be a wall of error text output, so it can be hard to spot the actual error amongst all the text.